cocotb/drivers/amba.py

```diff
@@ -351,22 +320,16 @@ def _read_data(self):
                     "Bytes in beat %d\n" % bytes_in_beat)
 
             burst_count = burst_length
-
-            yield clock_re
-
-            while True:
-                self.bus.RVALID <= 1
-                yield ReadOnly()
-                if self.bus.RREADY.value:
-                    _burst_diff = burst_length - burst_count
-                    _st = _araddr + (_burst_diff * bytes_in_beat)
-                    _end = _araddr + ((_burst_diff + 1) * bytes_in_beat)
-                    word.buff = self._memory[_st:_end].tostring()
-                    self.bus.RDATA <= word
-                    if burst_count == 1:
-                        self.bus.RLAST <= 1
-                yield clock_re
+            self.bus.RVALID <= 1
+            while burst_count > 0:
+                yield clock_re.do_until(lambda: self.bus.RREADY.value)
+                _burst_diff = burst_length - burst_count
+                _st = _araddr + (_burst_diff * bytes_in_beat)
+                _end = _araddr + ((_burst_diff + 1) * bytes_in_beat)
+                word.buff = self._memory[_st:_end].tostring()
+                self.bus.RDATA <= word
+                if burst_count == 1:
+                    self.bus.RLAST <= 1
                 burst_count -= 1
```

Comment:
This doesn't look right any more - `_burst_diff` is now always 1, whereas before it was not.

Reply:
I don't see why ``_burst_diff`` is always 1; ``burst_count`` is decreased for every iteration of the loop so ``burst_diff`` increases.